### PR TITLE
Modular dev create rpm tests

### PR DESCRIPTION
**Pull request by jkonecny12**
Move all nosetests to a separate folder.
Add RPM tests.
First part of RPM test is install test from Makefile.
Second part is comparing what is contained in rpm files and if everything is properly contained in our packages.
Add possibility to run just some RPM tests.
Grab logs to the result folder.

### CODE REVIEW

File: tests/rpm_tests/create_rpm_test.py [hunk 1]
```diff
@@ -0,0 +1,389 @@
+import os
```
Comment (poncovka) [inline tests/rpm_tests/create_rpm_test.py]: Please, add the usual header to the new files.

File: tests/rpm_tests/create_rpm_test.py [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+
+from tests.rpm_tests import RPMTestCase
+
+
+class InstallRPMTestCase(RPMTestCase):
```
Comment (poncovka) [inline tests/rpm_tests/create_rpm_test.py]: Please, add some comments to this class.

File: tests/rpm_tests/create_rpm_test.py [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+
+        src_files = self._apply_maps(
+            [
+                ModifyingFilters.remove_data_systemd_prefix,
+                lambda x: ModifyingFilters.apply_rpm_prefix("/usr/lib/systemd/system-generators",
+                                                            x),
+            ], src_files
+        )
+
+        src_files = self._check_and_return_as_list(src_files)
```
Comment (poncovka) [inline tests/rpm_tests/create_rpm_test.py]: You can probably call `_check_and_return_as_list` for `src_files` and `rpm_files` in the `_check_files_in_rpm` method.

Comment (jkonecny12) [reply tests/rpm_tests/create_rpm_test.py]: I don't think it is a good idea. It is then confusing that you see `src_files` checked but nothing for `rpm_files`.

If something like that then I'll rather create one check for both `src_files` and `rpm_files`.

**Comment by jkonecny12** [comment]
**UPDATED:**
* Implemented @poncovka suggestions.

File: tests/rpm_tests/create_rpm_test.py [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+                ModifyingFilters.remove_data_systemd_prefix,
+                lambda x: ModifyingFilters.apply_rpm_prefix("/usr/lib/systemd/system-generators",
+                                                            x)
+            ], src_files
+        )
+
+        self._check_files_in_rpm(src_files, rpm_files)
+
+    def _check_files_in_rpm(self, src_files, rpm_files):
+        self._check_and_return_as_list(src_files)
```
Comment (poncovka) [inline tests/rpm_tests/create_rpm_test.py]: Here should be probably `src_files = ...`

**Comment by jkonecny12** [comment]
**UPDATED:**
* Thanks for pointing that out @poncovka .

**Comment by jkonecny12** [comment]
jenkins, test this please

**Comment by jkonecny12** [comment]
jenkins, test this please

**Comment by jkonecny12** [comment]
jenkins, test this please

**Comment by jkonecny12** [comment]
jenkins, test this please

**Pull request merged**